### ISSUE TITLE
[Bug]: Dynamic api route does not pass query params

### ISSUE THREAD

**Issue by strefalinkow** [netlify/next-runtime#1726]
### Summary

Hi, after a week, I re-deployed the next.js app to netlify via auto deploy with the same code & dependency versions and I am not getting query parameters in a dynamic route.

My api route: `pages/api/post/[id]/delete.js`:
```
export default function handler(req, res) {
  console.log(req.query.id);
}
```
When I hit `/api/post/123/delete` the req.query.id is `"[id]"`.

Exactly the same next.js code deployed on netlify week later occurs this error.

So I specified the versions during the build (node, npm etc.) from the last successfull deploy, cleared the cache - nothing helped.
I also deployed the app on a different next.js hosting platform and this error does not occur.
I don't get any warnings or errors during build.

The last successful deployment dates back to October 10, where [4.24.3](https://github.com/netlify/next-runtime/compare/plugin-nextjs-v4.24.2...plugin-nextjs-v4.24.3) plugin-nextjs version was used by netlify.
Plugin-nextjs version used by netlify with this bug - [4.28.1](https://github.com/netlify/next-runtime/compare/plugin-nextjs-v4.28.0...plugin-nextjs-v4.28.1)

Have there been any changes between these versions that may cause this error?
On the netlify platform, I can only manually set the full versions of the plugin (1, 3, 4).

Thanks.

### Steps to reproduce

Log query param in a dynamic route.

### Next Runtime version

4.28.1

### More information about your build

- [ ] I am building using the CLI
- [ ] I am building using file-based configuration (`netlify.toml`)

### What OS are you using?

Windows

### Your netlify.toml file

<details>
  <summary>`netlify.toml`</summary>

```toml
# Paste content of your `netlify.toml` file here
```

</details>

### Your public/_redirects file

<details>
  <summary>`_redirects`</summary>

```toml
# Paste content of your `_redirects` file here
```

</details>

### Your `next.config.js` file

<details>
  <summary>`next.config.js`</summary>

```toml
const nextTranslate = require("next-translate");

module.exports = {
  images: {
    domains: ["firebasestorage.googleapis.com"],
  },
  reactStrictMode: true,
  ...nextTranslate(),
};

```

</details>

### Builds logs (or link to your logs)

<details>
  <summary>Build logs</summary>

```
# Paste logs here
```

</details>

### Function logs

<details>
  <summary>Function logs</summary>

```
# Paste logs here
```

</details>

### .next JSON files

<details>
  <summary>generated .next JSON files</summary>

```
# Paste file contents here. Please check there isn't any private info in them
# You can either build locally, or download the deploy from Netlify by clicking the arrow next to the deploy time.
```

</details>

**Comment by nickytonline** [comment]
Hi @strefalinkow, can you try version [4.26](https://github.com/netlify/next-runtime/releases/tag/plugin-nextjs-v4.26.0)? We had a big change in 4.27 (see https://github.com/netlify/next-runtime/pull/1495), so I just want to see if that's what's potentially impacting you.

**Comment by strefalinkow** [comment]
Yes, that was it. I specified the 4.26 plugin version in the `netlify.toml` file for deploy and everything works well.  The newer versions don't seem to work for me - I'm getting a pathname instead of a passed value in query params. Thanks for the help.

**Comment by ascorbic** [comment]
Yes, I think it may be that

**Comment by strefalinkow** [comment]
I just checked 4.28.3 - works.

**Comment by ascorbic** [comment]
Fixed in #1727
